python/cudf/cudf/core/column/column.py

```diff
@@ -414,14 +414,17 @@ def element_indexing(self, index):
         if val.null_count == 1:
             val = None
         else:
-            val = val.to_array()[0]
+            valid = (
+                cudautils.mask_get.py_func(self.mask_array_view, index)
```

Comment:
Maybe I'm missing something, but does the selection/`null_count` check above already handle this case?